x/bridge/keeper/transfers_test.go

```diff
@@ -0,0 +1,368 @@
+package keeper_test
+
+import (
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/osmosis-labs/osmosis/v23/app/apptesting"
+	"github.com/osmosis-labs/osmosis/v23/x/bridge/types"
+	tokenfactorytypes "github.com/osmosis-labs/osmosis/v23/x/tokenfactory/types"
+)
+
+func (s *KeeperTestSuite) TestInboundTransfer() {
+	accounts := apptesting.CreateRandomAccounts(10)
+
+	testCases := []struct {
+		name           string                   // test name
+		msg            types.MsgInboundTransfer // msg to send; NB! sender is set from the msgSenders slice
+		msgSenders     []string                 // sender of the msg; the msg is executed len(msgSenders) times
+		moduleSigners  []string                 // authorized module signers
+		assets         []types.Asset            // module assets
+		votesNeeded    uint64                   // votes needed module param
+		expectedErrors []error                  // expected errors, one per each msgSenders value
+		finalized      bool                     // expected finalization flag
+	}{
+		{
+			name: "transfer finalized: one signer, one vote",
+			msg: types.MsgInboundTransfer{
+				ExternalId:     "unique_external_id",
+				ExternalHeight: 100,
+				Sender:         "", // sender is set from the msgSenders slice
+				DestAddr:       accounts[9].String(),
+				AssetId:        assetID1,
+				Amount:         math.NewInt(100),
+			},
+			msgSenders: []string{
+				accounts[0].String(),
+			},
+			moduleSigners: []string{
+				accounts[0].String(),
+			},
+			assets:         []types.Asset{asset1},
+			votesNeeded:    1,
+			expectedErrors: []error{nil},
+			finalized:      true,
+		},
+		{
+			name: "transfer finalized: two signers, one vote needed",
+			msg: types.MsgInboundTransfer{
+				ExternalId:     "unique_external_id",
+				ExternalHeight: 100,
+				Sender:         "", // sender is set from the msgSenders slice
+				DestAddr:       accounts[9].String(),
+				AssetId:        assetID1,
+				Amount:         math.NewInt(100),
+			},
+			msgSenders: []string{
+				accounts[0].String(),
+				accounts[1].String(),
+			},
+			moduleSigners: []string{
+				accounts[0].String(),
+				accounts[1].String(),
+			},
+			assets:         []types.Asset{asset1},
+			votesNeeded:    1,
+			expectedErrors: []error{nil, nil},
+			finalized:      true,
+		},
+		{
+			name: "transfer finalized: one signer, two votes needed",
```

Comment:
I think name error here, should not be transfer finalized since two votes are needed